pyEpiabm/pyEpiabm/sweep/initial_place_sweep.py

```diff
@@ -0,0 +1,127 @@
+
+                if place.place_type.name in schools:  # schools
+                    # Initialise the fixed population
+                    helper.update_place_group(place, group_size=ave_group_size,
+                                              person_list=person_list,
+                                              person_weights=weights,
+                                              mean_capacity=mean_cap)
+
+                elif place.place_type.name == "Workplace":  # WORKSPACE
+                    # Fixed population is initialised on first run
+                    # thinking of making sure people don't have more than one
+                    # workplace
+                    power_list = [max_size, offset, power]
+                    helper.update_place_group(place, group_size=ave_group_size,
+                                              person_list=person_list,
+                                              person_weights=weights,
+                                              mean_capacity=mean_cap,
+                                              power_law_params=power_list)
+
+                elif place.place_type.name == "CareHome":  # CAREHOME
+                    # Kit will add more detail for Carehomes
+                    helper.update_place_group(place)
+
+        # Instantiate the temporary population in each place using
+        # the update sweep.
+        add_temporary_population = UpdatePlaceSweep()
+        add_temporary_population.bind_population(self._population)
+        add_temporary_population(0)
+
+    def create_age_weights(self, place, params):
+        """Function to return a list of people in the correct
+        age range to be added to a place, and the weights
+        for the different age groups.
+
+        Parameters
+        ----------
+
+        place : Place
+            Place to add people to
+        params : dict
+            Dictionary of parameters with age structure data
+
+        Returns
+        -------
+
+        typing.List[Person]
+            List of people who may be in the place
+        typing.List[float]
+            Corresponding weights for the person list
+
+        """
+        param_ind = place.place_type.value - 1
+        min_age = [params["age_group1_min_age"][param_ind],
+                   params["age_group2_min_age"][param_ind],
+                   params["age_group3_min_age"][param_ind]]
+        max_age = [params["age_group1_max_age"][param_ind],
+                   params["age_group2_max_age"][param_ind],
+                   params["age_group3_max_age"][param_ind]]
+        prop = [params["age_group1_prop"][param_ind],
+                params["age_group2_prop"][param_ind],
+                params["age_group3_prop"][param_ind]]
+
+        person_list = []
+        weights = []
+        for person in place.cell.persons:
+            if (place.place_type in person.place_types):
+                # People can't have more than one place of each type.
```

Comment:
Open Q - if we reach here, is that an error that should be logged, or just normal functionality we don't need to worry about? Does it matter if the place doesn't get as many people as it was meant to?

Reply:
I think normal functionality, because its looking at the whole cell each time, we wouldn't want to log as people are being perfectly normally put into workplaces etc and so being taken out of this list.